ipa-core/src/helpers/mod.rs

```diff
@@ -509,6 +509,19 @@ impl Display for TotalRecords {
 }
 
 impl TotalRecords {
+    /// Construct a `TotalRecords` specifying a record count of `value`.
+    ///
+    /// ## Panics
+    /// If the supplied record count is zero.
+    #[must_use]
+    pub const fn specified(value: usize) -> Self {
```

Comment:
This function is actually only used in one place, which calls it with a constant. I wasn't thinking clearly and mixed up "can be called in const context" with "must be called in const context". I'll just get rid of this one.

To be clear, you're looking for something that makes zero total records an error, rather than a panic? A bunch of the places `set_total_records` is used are in tests, where I'm not sure `?` is an easy way out. On the other hand, while `set_total_records` is used in lots of places, I'm more bothered by the clutter of lots of `unwrap`/`expect` than the work of updating them all.

Reply:
> To be clear, you're looking for something that makes zero total records an error, rather than a panic? 
That's right and only in production code. We can provide a function behind #[cfg(test)] that does panic and I think this is totally fine.

I do agree that `unwrap`/`expect` is just code clutter in this case and I was hoping that providing a function that returns `Result` will avoid it by using `?` - again only inside protocols and only if we need it. Does it make sense or I am missing something?

Reply:
Updated it to generate errors in protocol code.